Fixes outlined in pull #93

alichass opened the pull request:
- _build_sg_personalization "to" arg now changed to optional; either it or existing_personalizations must be set.
- Prevent using .personalizations without tos set.
- Prevent unnecessary resetting of tos when personalization is already constructed.

- No longer provide tos if not set for personalizations. This is to prevent every personalization from being sent to every recipient in msg.to

- Added/Changed some comments/docstrings to reduce ambiguity

sklarsa commented:
LGTM, @alichass can you add some test cases to validate the new logic?

To run the tests, check out the instructions [here](https://github.com/sklarsa/django-sendgrid-v5/blob/master/CONTRIBUTING.md)

alichass commented:
Hi I added the test to check the errors raised (the only new functionality I added really)  but for some reason when I run `nosetests -c nose.cfg` it doesnt seem to run the new tests (I checked this by adding an arbitrary assert that should have failed)

alichass commented:
![image](https://user-images.githubusercontent.com/31053605/132953068-85165d8b-ef1d-4347-88fd-1b3918ec0971.png)
